Vulkan-Pipeline-Assistant/shaderanalytics.h

```diff
@@ -15,6 +15,32 @@
 #include <QVulkanDeviceFunctions>
 
 namespace vpa {
+
+    struct PipelineConfig;
+
+    enum class ShaderStage {
+        VETREX, FRAGMENT, TESS_CONTROL, TESS_EVAL, GEOMETRY, count_
+    };
+
+    enum class SpirvResourceType {
+        INPUT_ATTRIBUTE, UNIFORM_BUFFER, STORAGE_BUFFER, SAMPLER_IMAGE, STORAGE_IMAGE, PUSH_CONSTANT
+    };
+
+    struct SpirvResource {
+        QString name;
+        size_t size;
+        SpirvResourceType resourceType;
+        SPIRV_CROSS_NAMESPACE::Resource spirvResource;
+
+        bool operator==(const SpirvResource& other) {
+            return name == other.name && size == other.size && resourceType == other.resourceType;
+        }
+        bool operator!=(const SpirvResource& other) {
+            return !operator==(other);
+        }
+    };
+
+
```

Comment:
Don't need to add in the enums and structs, they are defined in spirvresource.h. You only need the line 19 forward declaration, remove excess code.